### PR TITLE
Sample - ADF pipeline

### PR DESCRIPTION
**Pull request by balteravishay**
This PR adds a sample for using Azure Data Factory with the presidio-spark sample.

### CODE REVIEW

File: docs/samples/deployments/data-factory/index.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+## Pre-requisites
+
+### Azure Databricks
+
+Provision and setup the datbricks cluster by following the steps in [presidio-spark sample](../spark/index.md#Azure-Databricks). 
+**Note** that you should only create and configure the databricks cluster and not the storage account, which will be created in the next step.
+
+### Deploy the ARM template
+
+Create the rest of the service by running the following script which uses the provided ARM template.
```
Comment (navalev) [inline docs/samples/deployments/data-factory/index.md:23]:
```suggestion
Create the rest of the services by running the following script which uses the provided ARM template.
```

File: docs/samples/deployments/data-factory/index.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+```bash
+RESOURCE_GROUP=[Name of resource group]
+LOCATION=[location of resources]
+DATABRICKS_ACCESS_TOKEN=[Access token to databricks from the previous step]
```
Comment (navalev) [inline docs/samples/deployments/data-factory/index.md:28]:
```suggestion
DATABRICKS_ACCESS_TOKEN=[Access token to databricks created in the presidio-spark sample]
```

File: docs/samples/deployments/data-factory/index.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+DATABRICKS_WORKSPACE_URL=[Databricks workspace URL]
+DATABRICKS_CLUSTER_ID=[Databricks presidio-ready cluster ID]
+DATABRICKS_NOTEBOOK_LOCATION=[Location of presidio notebook from previous step]
```
Comment (navalev) [inline docs/samples/deployments/data-factory/index.md:31]:
```suggestion
DATABRICKS_NOTEBOOK_LOCATION=[Location of presidio notebook from the presidio-spark sample]
```

**Review by navalev — APPROVED** [review]

File: docs/samples/deployments/data-factory/azure-deploy.json [hunk 4]
```diff
@@ -0,0 +1,555 @@
...
+                                    "referenceName": "AzureKeyVault1",
+                                    "type": "LinkedServiceReference"
+                                },
+                                "secretName": "[parameters('AzureKeyVault_storageAccessKeySecretName')]"
+                            }
+                        }
+                    },
+                    "dependsOn": [
+                        "[parameters('dataFactoryName')]",
+                        "[concat(variables('factoryId'), '/linkedServices/AzureKeyVault1')]"
```
Comment (navalev) [inline docs/samples/deployments/data-factory/azure-deploy.json:162]: maybe use a more meaningfull name than AzureKeyVault1?

**Review by navalev — APPROVED** [review]

**Review by navalev — APPROVED** [review]

**Review by navalev — APPROVED** [review]

File: docs/samples/index.md [hunk 5]
```diff
@@ -10,3 +10,4 @@
 1. [Azure App service](deployments/app-service/index.md)
 2. [Kubernetes](deployments/k8s/index.md)
 3. [Spark/Azure Databricks](deployments/spark/index.md)
+4. [Azure Data Factory](deployments/data-factory/index.md)
```
Comment (navalev) [inline docs/samples/index.md:13]:
```suggestion
4. [Azure Data Factory with App Service](deployments/data-factory/index.md#pesidio-as-an-http-rest-endpoint)
5. [Azure Data Factory with Databricks](deployments/data-factory/index.md#pesidio-on-azure-databricks)
```

**Review by navalev — APPROVED** [review]

File: docs/samples/deployments/spark/presidio_anonymize_blobs.py [hunk 6]
```diff
@@ -38,6 +38,15 @@
...
 storage_account_access_key = dbutils.widgets.get("storage_account_access_key")
 
+
+# unount container if previously mounted
+def sub_unmount(str_path):
+    if any(mount.mountPoint == str_path for mount in dbutils.fs.mounts()):
+        dbutils.fs.unmount(str_path)
+
+
+sub_unmount("/mnt/files")
```
[the remark this answers is not in the document]
Comment (balteravishay) [reply docs/samples/deployments/spark/presidio_anonymize_blobs.py]: done

File: docs/samples/deployments/data-factory/index.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+The samples use the following Azure Services:
+
+* Azure Data Factory - Host and orchestrate the transformation pipeline.
+* Azure KeyVault - Holds the access keys for Azure Storage to avoid having keys and secrets in the code.
+* Azure Storage - Persistence layer of this sample.
+* Azure Databricks/ Azure App Service - Host presidio to anonymize the data.
+
+The input file used by the samples is hosted on [presidio-research](https://github.com/microsoft/presidio-research/) repository. It is setup as a variable on the provided ARM template and used by Azure Data Factory as the input source.
+
+## Presidio as an HTTP REST endpoint
```
Comment (omri374) [inline docs/samples/deployments/data-factory/index.md:17]:
```suggestion
## Option 1: Presidio as an HTTP REST endpoint
```

File: docs/samples/deployments/data-factory/index.md [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+az group create --name $RESOURCE_GROUP --location $LOCATION
+az deployment group create -g $RESOURCE_GROUP --template-file ./azure-deploy-adf-app-service.json
+```
+
+Note that:
+
+* A SAS token keys is created and read from Azure Storage and then imported to Azure Key Vault. Using ARM template built in [functions](https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/template-functions): [listAccountSas](https://docs.microsoft.com/en-us/rest/api/storagerp/storageaccounts/listaccountsas).
+* An access policy grants the Azure Data Factory managed identity access to the Azure Key Vault by using ARM template [reference](https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/template-functions-resource?tabs=json#reference) function to the Data Factory object and acquire its identity.principalId property. This is enabled by setting the data factory ARM resource's identity attribute to managed identity (SystemAssigned).
+
+## Presidio on Azure Databricks
```
Comment (omri374) [inline docs/samples/deployments/data-factory/index.md:38]:
```suggestion
## Option 2: Presidio on Azure Databricks
```

File: docs/samples/deployments/data-factory/index.md [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+* Azure KeyVault - Holds the access keys for Azure Storage to avoid having keys and secrets in the code.
+* Azure Storage - Persistence layer of this sample.
+* Azure Databricks/ Azure App Service - Host presidio to anonymize the data.
+
+The input file used by the samples is hosted on [presidio-research](https://github.com/microsoft/presidio-research/) repository. It is setup as a variable on the provided ARM template and used by Azure Data Factory as the input source.
+
+## Presidio as an HTTP REST endpoint
+
+By using Presidio as an HTTP endpoint, the user can select which infrastructure best suits their requirements. in this sample, Presidio is deployed to an Azure App Service, but other deployment targets can be used, such as [kubernetes](../k8s/index.md).
+
```
Comment (omri374) [inline docs/samples/deployments/data-factory/index.md:20]: Consider adding a screenshot of the ADF pipeline in each scenario.

File: docs/samples/deployments/data-factory/index.md [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+### Deploy the ARM template
+
+Create the Azure App Service and the ADF pipeline by running the following script, to provision the [provided ARM template](./azure-deploy-adf-app-service.json).
+
+```bash
+RESOURCE_GROUP=[Name of resource group]
+LOCATION=[location of resources]
+
+az group create --name $RESOURCE_GROUP --location $LOCATION
+az deployment group create -g $RESOURCE_GROUP --template-file ./azure-deploy-adf-app-service.json
```
Comment (omri374) [inline docs/samples/deployments/data-factory/index.md:30]: Can we do one-click-deploy for ADF?

**Review by omri374 — APPROVED** [review]
Looks great! awesome addition. Added some optional comments.

File: docs/samples/deployments/data-factory/index.md (hunk 7 above)
Comment (omri374) [reply docs/samples/deployments/data-factory/index.md:17]: If you accept this, make sure it doesn't break the link on the samples.md file.

**Review by navalev — DISMISSED** [review]

File: docs/samples/deployments/data-factory/index.md (hunk 10 above)
Comment (balteravishay) [reply docs/samples/deployments/data-factory/index.md:36]: done

File: docs/samples/deployments/data-factory/index.md (hunk 9 above)
Comment (balteravishay) [reply docs/samples/deployments/data-factory/index.md:20]: done

**Review by omri374 — APPROVED** [review]

**Review by navalev — APPROVED** [review]

**Pull request merged**